test/helpers/comment-issue-due-date.test.ts

```diff
@@ -0,0 +1,120 @@
+limitations under the License.
+*/
+
+import { Mocktokit } from '../types';
+import { context } from '@actions/github';
+import { commentIssueDueDate } from '../../src/helpers/comment-issue-due-date';
+import { octokit } from '../../src/octokit';
+import { PRIORITY_1, PRIORITY_2, PRIORITY_3, PRIORITY_4 } from '../../src/constants';
+
+jest.mock('@actions/core');
+jest.mock('@actions/github', () => ({
+  context: { repo: { repo: 'repo', owner: 'owner' }, issue: { number: 123 } },
+  getOctokit: jest.fn(() => ({
+    rest: { issues: { get: jest.fn(), createComment: jest.fn() } }
+  }))
+}));
+
+jest.spyOn(Date, 'now').mockImplementation(() => new Date('2023-09-26T10:00:00Z').getTime());
+
+describe('commentIssueDueDate', () => {
+  it('should add a due date of 2 days ahead when adding a critical priority label', async () => {
+    (octokit.issues.get as unknown as Mocktokit).mockResolvedValueOnce({
+      status: '200',
+      data: {
+        issue_number: 123,
+        created_at: Date.now(),
+        labels: [PRIORITY_1]
+      }
+    });
+    await commentIssueDueDate({ label: PRIORITY_1 });
+    expect(octokit.issues.createComment).toHaveBeenCalledWith({
+      body: 'This issue is due on Thu Sep 28 2023',
+      issue_number: 123,
+      ...context.repo
+    });
+  });
+  it('should add a due date of 14 days ahead when adding a high priority label', async () => {
+    (octokit.issues.get as unknown as Mocktokit).mockResolvedValueOnce({
+      status: '200',
+      data: {
+        issue_number: 123,
+        created_at: Date.now(),
+        labels: [PRIORITY_2]
+      }
+    });
+    await commentIssueDueDate({ label: PRIORITY_2 });
+    expect(octokit.issues.createComment).toHaveBeenCalledWith({
+      body: 'This issue is due on Tue Oct 10 2023',
+      issue_number: 123,
+      ...context.repo
+    });
+  });
+  it('should add a due date of 45 days ahead when adding a medium priority label', async () => {
+    (octokit.issues.get as unknown as Mocktokit).mockResolvedValueOnce({
+      status: '200',
+      data: {
+        issue_number: 123,
+        created_at: Date.now(),
+        labels: [PRIORITY_3]
+      }
+    });
+    await commentIssueDueDate({ label: PRIORITY_3 });
+    expect(octokit.issues.createComment).toHaveBeenCalledWith({
+      body: 'This issue is due on Fri Nov 10 2023',
+      issue_number: 123,
+      ...context.repo
+    });
+  });
+  it('should add a due date of 60 days ahead when adding a low priority label', async () => {
+    (octokit.issues.get as unknown as Mocktokit).mockResolvedValueOnce({
+      status: '200',
+      data: {
+        issue_number: 123,
+        created_at: Date.now(),
+        labels: [PRIORITY_4]
+      }
+    });
+    await commentIssueDueDate({ label: PRIORITY_4 });
+    expect(octokit.issues.createComment).toHaveBeenCalledWith({
+      body: 'This issue is due on Mon Dec 25 2023',
+      issue_number: 123,
+      ...context.repo
+    });
+  });
```

Comment:
These are all one test in my opinion. Just have one test that tests the happy path and one for the label not matching